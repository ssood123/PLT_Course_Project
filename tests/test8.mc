function void main3()
{
  int i;
  i = 2;
  print(i);
  main4();
}


function void main() 
{
  int i;
  i = 0;
  print(i);
  main2();
}

function void main2()
{
  int i;
  i = 1;
  print(i);
  main3();
}

function void main4()
{
  int i;
  i = 3;
  print(i);
}