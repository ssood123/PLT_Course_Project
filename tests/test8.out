; ModuleID = 'integraph'
source_filename = "integraph"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@fmt.3 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.4 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.5 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@fmt.6 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.7 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.8 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@fmt.9 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.10 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.11 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main4() {
entry:
  %i = alloca i32, align 4
  store i32 3, i32* %i, align 4
  %i1 = load i32, i32* %i, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 %i1)
  ret void
}

define void @main2() {
entry:
  %i = alloca i32, align 4
  store i32 1, i32* %i, align 4
  %i1 = load i32, i32* %i, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.3, i32 0, i32 0), i32 %i1)
  call void @main3()
  ret void
}

define void @main() {
entry:
  %i = alloca i32, align 4
  store i32 0, i32* %i, align 4
  %i1 = load i32, i32* %i, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.6, i32 0, i32 0), i32 %i1)
  call void @main2()
  ret void
}

define void @main3() {
entry:
  %i = alloca i32, align 4
  store i32 2, i32* %i, align 4
  %i1 = load i32, i32* %i, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.9, i32 0, i32 0), i32 %i1)
  call void @main4()
  ret void
}
